rust/lance/src/index/vector/ivf.rs

```diff
@@ -2226,15 +2231,11 @@ mod tests {
         let nlist = 4;
         let (mut dataset, vector_array) = generate_test_dataset(test_uri, 0.0..1.0).await;
 
-        let centroids = generate_random_array(nlist * DIM);
-        let ivf_centroids = FixedSizeListArray::try_new_from_values(centroids, DIM as i32).unwrap();
-        let ivf_params =
-            IvfBuildParams::try_with_centroids(nlist, Arc::new(ivf_centroids)).unwrap();
-
+        let ivf_params = IvfBuildParams::new(nlist);
         let pq_params = PQBuildParams::default();
         let hnsw_params = HnswBuildParams::default();
         let params = VectorIndexParams::with_ivf_hnsw_pq_params(
-            MetricType::L2,
+            MetricType::Cosine,
```

Comment:
should we create a new test for Cosine? Change it to cosine will lose converage on L2?

Reply:
the Cosine also covers L2 path, it's L2 + normalization

Reply:
we can keep one of the test using L2